test/core/local_set.wast

```diff
@@ -182,81 +182,17 @@
   "type mismatch"
 )
 
-(assert_invalid
-  (module (func $i32-vs-empty (param i32) (result i32) (local.set 0 (i32.const 1))))
```

Comment:
I just noticed that versions of these tests already existed and were removed in #933. Do you remember why? Can we restore the more explicit naming conventions they had before?

Reply:
Um, are you talking about those tests of `i*-vs-empty`? I don't see they were removed in #933, actually they were added in #918 and their names were suggested by you at https://github.com/WebAssembly/spec/pull/918#discussion_r236176421.

Reply:
Those `i*-vs-empty` are testing `local.set` should have no retval, and totally different from testing operand missing.  Seems it's a bit confusing on the name with "empty", how about something like "i32-vs-i32"?

Reply:
See the first batch of removals from local_set.wast in #933, multiple functions named $type-local-num-vs-num (for whatever reason). Aren't the ones here variations of the same?

Reply:
In a sense, `$type-local-num-vs-num` are the same with `i*-vs-empty`, that's why I removed them.
These tests fail for reasons completely unrelated to the type of the param or local, it's because of `local.set` should have no retval.
So, how about rename `i*-vs-empty` as "type-num-vs-num"? Or maybe we can just separate them with a standalone annotation something like ";; local.set should have no retval". WDYT?

Reply:
Thanks, I just rename them in new commit. PTAL.